Re-merged patch by Robert Scheck <robert@fedoraproject.org> for tcpick <= 0.2.1,
to fix multiple "implicitly converted to pointer" messages during build; patch is
originally from Dann Frazier <dannf@dannf.org>.

Further information:
 - http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=326927

--- tcpick-0.2.1/src/display.c			2004-12-31 14:53:30.000000000 +0100
+++ tcpick-0.2.1/src/display.c.pointers		2006-09-10 00:04:55.000000000 +0200
@@ -28,6 +28,7 @@
 /* FIXME: most code is duplicated: find a better solution! */
 
 #include "tcpick.h"
+#include "lookup.h"
 #include "extern.h"
 
 char * 
--- tcpick-0.2.1/src/tcpick.c			2005-01-18 13:25:36.000000000 +0100
+++ tcpick-0.2.1/src/tcpick.c.pointers		2006-09-10 00:06:02.000000000 +0200
@@ -49,6 +49,7 @@
 
 #include "tcpick.h"
 #include "globals.h"
+#include "datalink.h"
 
 char *errbuf[PCAP_ERRBUF_SIZE];
 struct bpf_program filter_compiled;
--- tcpick-0.2.1/src/write.c	2005-01-09 00:54:47.000000000 +0100
+++ tcpick-0.2.1/src/write.c.pointers	2006-09-10 00:06:47.000000000 +0200
@@ -25,8 +25,11 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
 
+#define _GNU_SOURCE
+#include <string.h>
 #include "tcpick.h"
 #include "extern.h"
+#include "lookup.h"
 
 __inline__ char * 
 avail_filename(struct CONN * conn_ptr, 
--- tcpick-0.2.1/src/lookup.h			2004-12-31 14:53:30.000000000 +0100
+++ tcpick-0.2.1/src/lookup.h.pointers		2006-09-10 00:07:38.000000000 +0200
@@ -26,6 +26,8 @@
  * USA.
  */
 
+#ifndef _LOOKUP_H
+#define _LOOKUP_H
 
 struct _l_node /* the node/leaf of the tree */
 {
@@ -40,3 +42,9 @@
 	char * name;
 	struct in_addr ip;
 };
+
+char *lookup(struct in_addr ia);
+struct _l_node *_l_alloc(struct in_addr, char *);
+char *_l_get(struct in_addr);
+
+#endif
--- tcpick-0.2.1/src/datalink.h			1970-01-01 01:00:00.000000000 +0100
+++ tcpick-0.2.1/src/datalink.h.pointers	2006-09-10 00:08:22.000000000 +0200
@@ -0,0 +1,29 @@
+/*
+ * datalink.h -- calculates datalink offsets
+ * Part of the tcpick project
+ *
+ * Author: Francesco Stablum <duskdruid @ despammed.com>
+ *
+ * Copyright (C) 2003, 2004  Francesco Stablum
+ * Licensed under the GPL
+ *
+ */
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at you option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111,
+ * USA.
+ */
+
+char *datalink2str(int dl_id);
